syntax= "proto3";
package parameterserver;
service ParameterServer {
  rpc PushGradients (GradientRequest) returns (WeightResponse) {}
}

message GradientRequest {
  string worker_id = 1;
  int32 step = 2;
  bytes gradient_data = 3;
  repeated int32 shape = 4;
}

message WeightResponse {
  bytes weight_data = 1;
  repeated int32 shape = 2;
}
